src/utils.ts

```diff
@@ -1240,22 +1248,30 @@ export const DEFAULT_PK_FACTORY = {
  * ```
  */
 export const MONGODB_WARNING_CODE = 'MONGODB DRIVER' as const;
+export const MONGODB_DEPRECATIONS: {
+  emittedWarnings: Set<string>;
+  emitWarning: (message: string) => void;
+} = Object.seal(
+  Object.assign(Object.create(null), {
+    emittedWarnings: new Set<string>(),
+    emitWarning: (message: string) => process.emitWarning(message, { code: MONGODB_WARNING_CODE })
```

Comment:
Actually, one thing that is nice about the `Logger` API is that it is configurable on a per-MongoClient basis, not mutable global state. This PR isn't really changing that, but if there's an obvious way to make that happen, that would probably be quite a bit more useful for mongosh.